### DISCUSSION TITLE
Zephyr Linker modification to use SRAM_2

### DISCUSSION THREAD

**Discussion by AlbertPJoshi** [zephyrproject-rtos/zephyr#70715] [Q&A]
While building an application my SRAM area is overflowed. During the investigation i noticed that in the current set up(based on the post build prints and linker scripts, .map file) zephyr not using the SRAM_2(my MCU have two separate RAM regions called SRAM_1 and SRAM_2, it is not continuous memory)

My MCU's SRAM allocation is :
SRAM1 - 32 KB(this one is currently using by zephyr)
SRAM2-  30KB(It is not used by the zephyr)

So in order to use this (unused SRAM_2)area i have tried modifying the linker using CONFIG_CODE_DATA_RELOCATION(in project.conf), but even though adding this and some proposed changes on the CMakeList.txt i was not able to use the SRAM_2 area.

Is there any way to combine these two RAM areas(it is not continuous as i mentioned).?
Or
What is the proper way to modify the linker in order to use the SRAM_2 and how to allocate memory to that region?

**Comment by smo591** [comment]
Did you ever figure out how to do this?
I have the same problem with an STM32H7 MCU that has 6 different blocks of SRAM but only 3 of them are contiguous.

**Reply by AlbertPJoshi** [reply]
If contigous treat it as a single block and alloacte some variables at the border aaddresses, to prevent burst access.
Seems to be holding pretty well,
But if non-contigous you will have to modify linker I, guess?

**Comment by KozhinovAlexander** [comment]
Are you using ethernet?

**Reply by AlbertPJoshi** [reply]
Nope

**Reply by KozhinovAlexander** [reply]
Please take a look on stm32h7 ethernet driver buffer reallocation with modification of the linker script here: https://github.com/zephyrproject-rtos/zephyr/commit/c276088567b500286bc613768cf7542552f3eda1#diff-73fdb3599af069c4d058668061236c555fc1007c5faf040a08b414cf8aa3c88e 

The above PR is introducing exactly, what you mean. This is only technique I know within zephyr, which were used to overcome exactly same limitation you mentioned.
